pkg/identityserver/mock/gateway_registry.go

```diff
@@ -42,22 +60,13 @@ func (is *mockISGatewayRegistry) SetRegisteredGateway(gtwIDs *ttnpb.GatewayIdent
 	is.registeredGateway = gtwIDs
 }
 
-func (is *mockISGatewayRegistry) Add(ctx context.Context, ids ttnpb.GatewayIdentifiers, key string, locationPublic, updateLocationFromStatus bool, rights ...ttnpb.Right) {
+// Add gateway to the mock registry. If gateway is not specified the mockis.DefaultGateway will be used
+func (is *mockISGatewayRegistry) Add(ctx context.Context, ids ttnpb.GatewayIdentifiers, key string, gtw *ttnpb.Gateway, rights ...ttnpb.Right) {
 	uid := unique.ID(ctx, ids)
-	is.gateways[uid] = &ttnpb.Gateway{
-		Ids:              &ids,
-		FrequencyPlanId:  test.EUFrequencyPlanID,
-		FrequencyPlanIds: []string{test.EUFrequencyPlanID},
-		Antennas: []*ttnpb.GatewayAntenna{
-			{
-				Location: &ttnpb.Location{
-					Source: ttnpb.LocationSource_SOURCE_REGISTRY,
-				},
-			},
-		},
-		GatewayServerAddress:     "mockgatewayserver",
-		LocationPublic:           locationPublic,
-		UpdateLocationFromStatus: updateLocationFromStatus,
+	if gtw == nil {
```

Comment:
Nit: I would say that it is better if the call sites call `DefaultGateway` explicitly.